Disable Run > Files > Tale Workspaces UI if insufficient Tale access

bodom0015 opened the pull request:
### Problem
User can launch a Tale to which they don't have sufficient read or write access. Tale Workspaces UI becomes non-functional in this case, since we can't read the Workspace or upload files/folders to it.

Fixes #391

### Approach
Disallow user from selecting Tale Workspaces without read access, disallow Add button without write access

### How to Test
1. Checkout and run this branch locally
2. Login to the WholeTale Dashboard
3. Launch a Tale that does not allow you read access
4. Navigate to Run > Files
    * You should see that Tale Workspaces has been greyed out, disallowing reads
5. Launch a Tale that allows you to read, but not write
6. Navigate to Run > Files > Tale Workspaces
    * You should see the Tale's Workspace folders/files are listed here
    * You should see that (+) has been greyed out, disallowing writes

craig-willis commented:
Per 2/18 discussion, when testing with https://github.com/whole-tale/girder_wholetale/pull/243, the underlying problem fixed here is unlikely, but the PR is still valid.  More important is correct handling of the read-only case (including public=true).  If the Tale/Workspace is read-only, the user should be able to view the contents of the workspace, download and copy items, but not write, upload, rename, create folder, remove, or move, etc.

bodom0015 commented:
I've updated the PR to handle the "Read-only Tale" case a bit better:
* Ensured that the Tale Workspaces nav is enabled based on both `accessLevel` and `public`
* Ensured that the (+) button is disabled appropriately
* Removed all folder/file dropdown options except for "Copy to Home" and "Download"

Xarthisius commented:
LGTM! I found one issue, but I don't know if it should be addressed in this PR (or at all):
It's possible to try to move files from ro workspaces which results in error shown below:
![workspace](https://user-images.githubusercontent.com/352673/53030533-95ee2e80-3430-11e9-8a74-716e313c3fd2.png)

Do we want to add logic checking if source workspace is ro and gray out "Move to Workspace" button?
